### DISCUSSION TITLE
v2.71.1 - main.js

### DISCUSSION THREAD

**Discussion by VaclavElias** [dotnet/docfx#9332] [Q&A] [answered]
Hi,

The most recent update seems to initialise `main.js` in a different way, and so I can't use this approach below

https://github.com/stride3d/stride-docs/blob/master/en/template/public/main.js

```
// Start the app when the DOM content is loaded
document.addEventListener("DOMContentLoaded", () => app.start());
```

Causing now

![image](https://github.com/dotnet/docfx/assets/4528464/db538c32-8246-4d45-a04a-875b5aadca49)

Any alternative suggestion, how can I trigger my start function, without creating a custom template?

Thank you.

**Comment by yufeih** [comment]
Arrr, the `search-worker.ts` also imports `main.js` for `configureLunr`, as workaround, check if the script is running in web worker: https://stackoverflow.com/a/11979803

**Comment by VaclavElias** [comment]
I have fixed my issue by calling `start()` like this:

```
app.start = app.start.bind(app);
```

**Reply by yufeih** [reply, marked as answer]
Now there is a dedicated [`start`](https://github.com/dotnet/docfx/blob/main/templates/modern/src/options.d.ts#L42) callback in `main.js` to handle `start` event:

```js
export default {
  start() {
    // start logic
  }
}
```

**Reply by VaclavElias** [reply]
Thanks. I am now utilising it.

**Reply by dennisfike** [reply]
Thank-you so much VaclaElias. I was looking to just add in a li in the navbar to just display a currentDayMessage  if you will. I was trying add this item for hours until I came across your repo. I better get with the program with observables. That's for sure.
